client/client-multi/client-c/listfiles/UnitTest.cmake

```diff
@@ -26,154 +26,203 @@ if(CUNIT_FOUND)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DKAA_TEST_CUNIT_ENABLED")
 endif()
 
+################################################################################
+# Creates an unit test with given name and dependencies.
+#
+# TODO: move this function to the separate module, after new design will be implemented.
+#       When new SDK build system will be ready, tests will be encapsulated
+#       within each production module. Test system helpers, like this function
+#       will be moved to separate cmake build modules or scripts.
+#
+# Syntax:
+# kaa_add_unit_test(NAME test_name
+#                   SOURCES test_sources_files...
+#                   [DEPENDS list_of_dependencies...]
+#                   [INC_DIRS list_of_include_directories...])
+function(kaa_add_unit_test)
+    cmake_parse_arguments(
+            UNIT_TEST
+            "OPTIONAL"
+            "NAME"
+            "SOURCES;DEPENDS;INC_DIRS"
+            ${ARGN}
+    )
+
+    if (DEFINED UNIT_TEST_NAME AND DEFINED UNIT_TEST_SOURCES)
```

Comment:
You can use `if (NOT DEFINED UNIT_TEST_NAME OR NOT DEFINED UNIT_TEST_SOURCES)` and abort then. Less nesting